src/current_input.rs

```diff
@@ -101,10 +104,11 @@ impl CurrentInput {
 
                 match delta {
                     MouseScrollDelta::LineDelta(_, y) => {
-                        self.scroll_diff += y;
+                        self.y_scroll_diff += y;
```

Comment:
I think this needs to handle x_scroll_diff as well

Reply:
Added that